src/erlpmd_ets.erl

```diff
@@ -36,19 +42,20 @@ node_port(NodeName, erlpmd) ->
 
 names(erlpmd) ->
     {ok, [{Name, Port}
-          || [Name, Port] <- ets:match(erlpmd, {'$1', {'$2', 77, '_', '_', '_', '_', '_', '_'}})]}.
+          || [Name, Port] <- ets:match(erlpmd, #node{name='$1', port_no='$2', type=77, _='_'})]}.
 
 dump(all, erlpmd) -> dump('_', erlpmd);
 dump(NodeType, erlpmd) ->
     {ok, [{Name, Port, Fd}
           || [Name, Port, Fd] <- ets:match(erlpmd,
-                                           {'$1', {'$2', NodeType, '_', '_', '_', '_', '$3', '_'}})]}.
+                                           #node{name='$1', port_no='$2', type=NodeType, fd='$3', _='_'})]}.
 
 node_stopped(Fd, erlpmd) ->
-	case ets:match(erlpmd, {'$1', {'_', '_', '_', '_', '_', '_', '_', Fd}}) of
+	case ets:match(erlpmd, #node{name='$1', fd=Fd, _='_'}) of
```

Comment:
I think this might have been a bug. `Fd` is second last when the code is inserted.